plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClientModule.java

```diff
@@ -53,34 +59,89 @@ public void configure(Binder binder)
         configBinder(binder).bindConfig(OracleConfig.class);
         newOptionalBinder(binder, Key.get(int.class, MaxDomainCompactionThreshold.class)).setBinding().toInstance(ORACLE_MAX_LIST_EXPRESSIONS);
         newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, RetryStrategy.class).setBinding().to(OracleRetryStrategy.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, Key.get(ConnectionFactory.class, ForRetryingConnectionFactory.class))
+                .setBinding()
+                .toProvider(OracleConnectionFactoryProvider.class)
```

Comment:
I don't get how does it work with `@ForBaseJdbc` today. Until now specific connector was required to provide `@ForBaseJdbc ConnectionFactory`, now it needs to provide `@ ForRetryingConnectionFactory ConnectionFactory`. How do they relate to each other? How do they build the ultimate `ConnectionFactory` that is used to create the real connection?

Reply:
That was the 'previous' approach before we changed the acceptance criteria. This code is now deleted.